handler/core/strategy/hmacsha.go

```diff
@@ -0,0 +1,36 @@
+package strategy
+
+import (
+	"github.com/ory-am/fosite"
+	"github.com/ory-am/fosite/enigma"
+	"golang.org/x/net/context"
+	"net/http"
+)
+
+type HMACSHAStrategy struct {
+	Enigma enigma.HMACSHAEnigma
+}
+
+func (h HMACSHAStrategy) GenerateAccessToken(_ context.Context, _ *http.Request, requester fosite.AccessRequester, _ interface{}) (token string, signature string, err error) {
+	return h.Enigma.Generate(requester.GetClient().GetHashedSecret())
```

Comment:
by the way i think it would be okay to remove the context variable as I'm not sure that the generators/ validators really need it. however, I do not want to change the interface after this PR here any more so I can't really decide if keep or kick :D